Update the homepage and section indexs of docs (#187)
canonical/httprequest-lego-provider: charm/docs/index.rst

.. Copyright 2026 Canonical Ltd.
.. See LICENSE file for licensing details.

.. meta::
   :description: Discover the HTTP request Lego provider charm, a Juju operator that deploys and manages HTTP request Lego provider.

.. vale Canonical.007-Headings-sentence-case = NO

.. _index:

HTTP request Lego provider operator
===================================

.. vale Canonical.007-Headings-sentence-case = YES

> A `Juju <https://juju.is/>`_ `charm <https://documentation.ubuntu.com/juju/3.6/reference/charm/>`_

Reviewer: ```suggestion
A `Juju <https://juju.is/>`_ `charm <https://canonical.com/juju/docs/juju-cli/3.6/reference/charm/>`_
```

deploying and managing HTTP request Lego provider on Kubernetes. 
The HTTP request Lego provider is a web application implementing the DNS HTTP request provider as defined by ACME to manage DNS records.

Like any Juju charm, this charm supports one-line deployment, configuration, integration,
scaling, and more. 
For HTTP request Lego provider, this includes:

* `ACME DNS-01 challenge <https://letsencrypt.org/docs/challenge-types/#dns-01-challenge>`_ support for TLS certificate issuance and renewal.
* A simple web interface for managing DNS records.
* Admin and user roles for managing access to DNS records.

The HTTP request Lego provider charm allows for deployment on many different Kubernetes platforms,

> from `MicroK8s <https://microk8s.io/>`_ to 

Reviewer: ```suggestion
from `MicroK8s <https://canonical.com/microk8s>`_ to 
```

.. Copyright 2026 Canonical Ltd.
.. See LICENSE file for licensing details.

.. meta::
   :description: Discover the HTTP request Lego provider charm, a Juju operator that deploys and manages HTTP request Lego provider.

.. vale Canonical.007-Headings-sentence-case = NO

.. _index:

HTTP request Lego provider operator
===================================

.. vale Canonical.007-Headings-sentence-case = YES

A `Juju <https://juju.is/>`_ `charm <https://documentation.ubuntu.com/juju/3.6/reference/charm/>`_
deploying and managing HTTP request Lego provider on Kubernetes. 
The HTTP request Lego provider is a web application implementing the DNS HTTP request provider as defined by ACME to manage DNS records.

Like any Juju charm, this charm supports one-line deployment, configuration, integration,
scaling, and more. 
For HTTP request Lego provider, this includes:

> * [ACME DNS-01 challenge](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge) support for TLS certificate issuance and renewal.

Reviewer: ```suggestion
* `ACME DNS-01 challenge <https://letsencrypt.org/docs/challenge-types/#dns-01-challenge>`_ support for TLS certificate issuance and renewal.
```
since this is a reStructuredText file, links are formatted differently

* A simple web interface for managing DNS records.
* Admin and user roles for managing access to DNS records.

The HTTP request Lego provider charm allows for deployment on many different Kubernetes platforms,
from `MicroK8s <https://microk8s.io/>`_ to 
`Charmed Kubernetes <https://ubuntu.com/kubernetes>`_ to public cloud Kubernetes offerings.

This charm will make managing and renewing TLS certificates for your applications straightforward.

In this documentation
---------------------

.. list-table::
    :header-rows: 0

    * - Get started

>       - :doc:`Tutorial <tutorial/getting-started>`

Reviewer: IMO it's good for us to provide some indication of what users will learn in the tutorial, maybe something like:
```suggestion
      - :doc:`Deploy and access HTTP request Lego provider <tutorial/getting-started>`
```

    * - Operations
      - :doc:`Back up and restore <how-to/backup-and-restore>` | :doc:`Troubleshoot API timeouts <how-to/troubleshoot-api-timeouts>` | :doc:`Upgrade <how-to/upgrade>`
    * - Product-specific feature

>       - :doc:`Manage domains <how-to/manage-domains>` | :doc:`Manage users <how-to/manage-users>`

Reviewer: "Product-specific feature" is a placeholder term. What specific feature are you highlighting for the charm here? 

IMO these seem like "Deployment" level guides, i.e., things we can expect users to do in Days 0-1. What do you think?

Author: Done

    * - Design

>       - :doc:`Charm architecture <reference/charm-architecture>`

Reviewer: I would put the "Integrations" page in this row

Author: Done.

    * - Security

>       - :doc:`Security <explanation/security>`

Reviewer: Nit: To avoid duplicating "security" twice, we could rephrase as:
```suggestion
    * - Security
      - :doc:`Overview <explanation/security>`
```

How this documentation is organized
------------------------------------

This documentation uses the `Diátaxis documentation structure <https://diataxis.fr/>`_.

- The :doc:`Tutorial <tutorial/index>` takes you step-by-step through a basic deployment of the <Charm software> charm.

> - The :doc:`How-to guides <how-to/index>` assume you have basic familiarity with the <Charm software> charm. Learn more about setting up, using, maintaining, and contributing to this charm.

Reviewer: ```suggestion
- The :doc:`Tutorial <tutorial/index>` takes you step-by-step through a basic deployment of the HTTP request Lego provider charm.
- The :doc:`How-to guides <how-to/index>` assume you have basic familiarity with the charm. Learn more about setting up, using, maintaining, and contributing to this charm.
```

- The :doc:`Reference <reference/index>` provides a guide to actions, configurations, relations, and other technical details.

> - The :doc:`Explanation <explanation/index>` includes topic overviews, background and context and detailed discussion.

Reviewer: Since you added targets for all the index pages, I think you should use them instead of absolute paths here (i.e., convert `:doc:` to `:ref:` and use the index page targets for cross-referencing)

Author: I have updated to use :ref: now. Can you check if this is what you are expecting?

Contributing to this documentation
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Documentation is an important part of this project, and we take the same open-source approach
to the documentation as the code. As such, we welcome community contributions, suggestions, and
constructive feedback on our documentation.
See our `contributing guide
<https://github.com/canonical/httprequest-lego-provider/blob/main/CONTRIBUTING.md>`_
for more information.

If there's a particular area of documentation that you'd like to see that's missing, please 
file a bug.

> .. TODO: Add link to GitHub issues page for "file a bug"

Reviewer: ```suggestion
If there's a particular area of documentation that you'd like to see that's missing, please 
`file a bug <https://github.com/canonical/httprequest-lego-provider/issues/>`_.
```
